tutorials/raw/plot_object_raw.py

```diff
@@ -2,131 +2,557 @@
-# ------------------------------------------
-# Concatenating :class:`~mne.io.Raw` objects
-# ------------------------------------------
+# The ``Raw.get_data()`` method
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 #
-# :class:`~mne.io.Raw` objects can be concatenated in time by using the
-# :func:`~mne.io.Raw.append` function. For this to work, they must
-# have the same number of channels and their :class:`~mne.Info`
-# structures should be compatible.
+# If you only want the data (not the corresponding array of times),
+# :class:`~mne.io.Raw` objects have a :meth:`~mne.io.Raw.get_data` method. Used
+# with no parameters specified, it will return all data from all channels, in a
+# (n_channels, n_timepoints) :class:`NumPy array <numpy.ndarray>`:
+
+data = raw.get_data()
+print(data.shape)
 
-# Create multiple :class:`~mne.io.Raw` objects
-raw1 = raw.copy().crop(0, 10)
-raw2 = raw.copy().crop(10, 20)
-raw3 = raw.copy().crop(20, 40)
+###############################################################################
+# If you want the array of times, :meth:`~mne.io.Raw.get_data` has an optional
+# ``return_times`` parameter:
+
+data, times = raw.get_data(return_times=True)
+print(data.shape)
+print(times.shape)
 
-# Concatenate in time (also works without preloading)
-raw1.append([raw2, raw3])
-print('Time extends from', raw1.times.min(), 's to', raw1.times.max(), 's')
+###############################################################################
+# The :meth:`~mne.io.Raw.get_data` method can also be used to subselect
+# specific channel(s) and sample ranges, via its ``picks``, ``start``, and
+# ``stop`` parameters. The ``picks`` parameter accepts integer channel indices,
+# channel names, or channel types, and preserves the requested channel order
+# given as its ``picks`` parameter.
+
+first_channel_data = raw.get_data(picks=0)
+eeg_and_eog_data = raw.get_data(picks=['eeg', 'eog'])
+two_meg_chans_data = raw.get_data(picks=['MEG 0712', 'MEG 1022'],
+                                  start=1000, stop=2000)
+
+print(first_channel_data.shape)
+print(eeg_and_eog_data.shape)
+print(two_meg_chans_data.shape)
+
+###############################################################################
+# Summary of Raw indexing techniques
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+# The following table summarizes the various ways of subselecting data from a
+# :class:`~mne.io.Raw` object.
+#
+# .. cssclass:: table-bordered
+# .. rst-class:: midvalign
+#
+# +-------------------------------------+-------------------------+-----------+
+# | Python code                         | Result                  | Modifies  |
+# |                                     |                         | Raw       |
+# |                                     |                         | in-place? |
+# +=====================================+=========================+===========+
+# | ``raw.get_data()``                  | :class:`NumPy array     |           |
```

Comment:
I wouldn't call this indexing since there is no "index" that is passed to get_data

Reply:
can you suggest an alternate title for the table then?